apps/desktop/src/renderer/screens/main/components/WorkspaceView/ContentView/TabsContent/Terminal/hooks/useTerminalConnection.ts

```
@@ -15,6 +16,57 @@ export interface UseTerminalConnectionOptions {
+
+function utf8Hex(data: string): string {
+	return Array.from(new TextEncoder().encode(data))
+		.map((byte) => byte.toString(16).padStart(2, "0"))
+		.join(" ");
+}
+
+function summarizeTerminalInput(data: string): Record<string, unknown> {
+	return {
+		bytes: new TextEncoder().encode(data).length,
+		chars: data.length,
+		escapes: (data.match(/\x1b/g) || []).length,
```

Biome (2.4.11)

[error] Unexpected control character in a regular expression.

(lint/suspicious/noControlCharactersInRegex)